Automattic/newspack-plugin
# Plugin Installer Component

### All Submissions:






### Changes proposed in this Pull Request:

This PR adds a new component that provides a UI for installing plugins. The component takes one parameter, `plugins`, which is an array of the plugin slugs to display and manage. Assuming those plugins have been defined/whitelisted in https://github.com/Automattic/newspack-plugin/blob/8cced0e613fafaa6aef086ff13f4c6acb11d8f22/includes/class-plugin-manager.php, the component displays information about each one in a separate `ActionCard`. The UI allows for installation/activation (these are treated as a single action with no distinction), deinstallation, and one-click installation of all listed plugins. If installation fails, an error notification will be posted. I added a fake plugin that is guaranteed to fail as a demonstration of the notifications. 

### How to test the changes in this Pull Request:

1. Check out the branch.
2. `npm ci && npm run clean && npm run build:webpack`
3. Navigate to Newspack->Components Demo, observe Plugin Installer at the top of the page with four plugins listed (AMP, WooCommerce, Yoast SEO, "Fake Plugin")
4. Click `Install` button for individual plugins and verify that they are successfully installed.
5. When clicking `Install` for Fake Plugin, observe that installation fails and an error notification is posted with the text "Plugin in not found."
6. Hover over cards and click `Uninstall`, verify that plugins are deactivated successfully. (Note they will not be deleted, just deactivated)
7) Deactivate all plugins, then click `Install All`. Verify that all plugins are installed and activated, except for Fake Plugin which will fail.

<img width="509" alt="Screen Shot 2019-05-30 at 6 35 47 PM" src="https://user-images.githubusercontent.com/1477002/58669314-c84dd880-8309-11e9-9809-7d2db761d51d.png">

### Other information:
